e2ev2/src/circle-run-tests.sh

```diff
@@ -31,8 +31,7 @@ set +e
 export FAILED_TESTS_LOG=failed-tests.txt
 # This should do the right thing. If $FAILED_TESTS is empty, nothing is specified, so Jest runs
 # all tests.
-yarn test $FAILED_TESTS \
-  --reporters=jest-silent-reporter --reporters=./src/failure-reporter.js
```

Comment:
Failure reporter is needed to ensure re-runs don't run every test again. I think you should revert this and we can isolate it into its own PR.